chainer/utils/type_check.py

```diff
@@ -486,4 +489,17 @@ def expect(*bool_exprs):
         assert isinstance(expr, Testable)
         expr.expect()
 
+
+def same_types(*arrays):
+    are_numpy_arrays = map(lambda x: issubclass(type(x), numpy.ndarray),
+                           arrays)
+    all_numpy_arrays = all(are_numpy_arrays)
+    if cuda.available:
+        are_cupy_arrays = map(lambda x: issubclass(type(x), cupy.ndarray),
```

Comment:
Note that we must not remove `cuda.available` here because `cupy.ndarray` is not available when `cuda.available` is `False`.